### PR TITLE
Views: Entity Dropdown

### PR DESCRIPTION
**Pull request by gblakeman**
Adds a drop-down visual menu (`PillList`) for entities in the `PillActionPrimary` component. The new components are used in the context of the larger cards in the UI (`CardPrimary`).

`PillList` shares a lot of similarities (and some code) with `ItemsList`, so this round of work also fixes a number of outstanding, smaller issues in the `ItemsList`, including:

* Close when clicking outside the layers tray
* Auto-scroll the layer list to the focused item on re-opening

----

### CODE REVIEW

File: src/Presenter.ts [hunk 1]
```diff
@@ -197,26 +214,13 @@ const setEntPstItem = (
       }
     });
 
-    // assemble list for dropdown
```
Comment (gblakeman) [inline src/Presenter.ts:200]: The drop-down menu list was already included in the bundle sent to the UI thread, but in the future, it will be possible for each large pill component to have its own list, so I moved the code that generates the list into the same block that generates the view bundle for the large pills.

File: src/Tools.ts [hunk 2]
```diff
@@ -41,6 +41,36 @@ const addOrRemoveItemId = (
   return updatedItemsArray;
 };
 
+/**
+ * Watches for clicks outside of a supplied `HTMLElement`. If an outside click
```
Comment (gblakeman) [inline src/Tools.ts:45]: The way this is formatted is specific to use in a Svelte component. Might be a good candidate for the Shared Tools package.

File: src/views/ItemsList.svelte [hunk 3]
```diff
@@ -344,15 +361,17 @@
...
-        <Item
-          on:handleItemClick={() => setSelection(item, i, items)}
-          assignment={item.contentInfo.assignment}
-          assignmentType={item.contentInfo.id ? 'persona' : 'detached'}
-          focused={item.id === $focusedItem.id}
-          nodeType={item.nodeType}
-          name={item.name}
-          selected={isSelected(item.id, $selectedItems)}
-        />
+        <li>
```
Comment (gblakeman) [inline src/views/ItemsList.svelte:364]: Had a list without actual list items for quite awhile now… oops! 🤦🏻‍♂️

File: src/Tools.ts [hunk 4]
```diff
@@ -263,8 +295,73 @@ const resizeGUI = (
   );
 };
 
+/**
+ * Scrolls a list view to the focused element. Requires a component to have
+ * a scrolling element that is bound to `this` in the markup of the scrollable
+ * container: `bind:this={elementName}`.
+ *
+ * @kind function
+ * @name setListScroll
```
Comment (gblakeman) [inline src/Tools.ts:304]: Another possible Shared Tools contender if we thought it would get used in other plugins.

File: src/code.ts [hunk 5]
```diff
@@ -102,6 +102,12 @@ const dispatcher = async (action: {
         App.refreshContentItem(assignment, childId);
         break;
       }
+      case 'content-load-item': {
+        const { id, parentId }: {
+          id: string, parentId: RealishAssignmentEntPst } = payload;
```
Comment (lynnerang) [inline src/code.ts:107]: minor thing - took me a sec to read this and wondered if the linter might want a line break before the closing bracket?

**Review by lynnerang — APPROVED** [review]
👍 🙌

File: src/code.ts (hunk 5 above)
Comment (gblakeman) [reply src/code.ts:107]: Linter seems cool with it either way, but I can see how it’s hard to follow. I’ll add the line break.
